db/migrate/20200524080344_create_storage_systems.rb

```diff
@@ -0,0 +1,17 @@
+class CreateStorageSystems < ActiveRecord::Migration[5.1]
+  def change
+    create_table :storage_systems do |t|
```

Comment:
We have an existing `PhysicalStorage` model, `StorageSystem` sounds like the same concept so I don't want to introduce a duplicate model.  Can you add any attributes that you need which are missing to the physical_storages table instead?

Reply:
The motivation for creating a new table was that many of the relations and in the physical_storage are not relevant to other storages (rack, chassis, canisters, computer_system,  ...). The propose a design of a storage_system that is general / minimalistic table that includes only the common fields needed for storage system and relations to only general objects common to all storage systems (i.e. Volumes, Ports) . Each type of storage may inherit from this table and add it's own personal relations and fieles. . This way each field and relation in the table is populated with relevant info and we wont have many empty non relevant fields or relation for each storage type.

Reply:
Are these `storage_systems` going to be representing actual physical storage systems with hard drives etc...?   The management_ip makes me think that they are.

If the issue is that autosde won't be populating rack/chassis and lenovo won't be populating storage-pools that's perfectly fine, no provider populates every association on the models that it collects and we can always not show relationships that aren't populated.

I think this will be more confusing when we start adding other standalone storage managers, e.g. if we added a NetApp provider that does have physical_storages but also storage-pools, would it populate both physical_storages and storage_systems?

If it would help we could move the associations to e.g. `physical_racks` and `physical_chassis` into a `PhysicalInfraManager::PhysicalStorage` subclass so that your provider's subclass of `PhysicalStorage` wouldn't have them.

Reply:
So looking at https://en.wikipedia.org/wiki/IBM_Storwize_family#Storwize_V7000_Gen_3_(or_'Next_Gen') it has drive bays/enclosures (both on physical_storage table) and multiple canisters (a physical_storage has_many canisters)

As for https://en.wikipedia.org/wiki/IBM_SAN_Volume_Controller that almost sounds like its own standalone storage_manager type which aggregates many physical_storages the same way vCenter manages multiple ESXi hosts

Reply:
Maybe we rename physical_storages to storage_systems since that is a good name for a general storage type, then physical storages can be a subclass of that?

Reply:
@agrare, We'll move everything into PhysicalStorage.